envi/archs/arm/disasm.py

```diff
@@ -566,12 +593,20 @@ def p_dp_reg_shift(opval, va):
     Rm = opval & 0xf
     shtype = (opval >> 5) & 0x3
     Rs = (opval >> 8) & 0xf
-
+    mnem = dp_mnem[ocode]
     if ocode in dp_noRn:# FIXME: FUGLY
-        olist = (
-            ArmRegOper(Rd, va=va),
-            ArmRegShiftRegOper(Rm, shtype, Rs),
-        )
+        #no register shift displayed with mov
+        if  (ocode == 13):
+            mnem = dp_shift_mnem[shtype]
+            olist = (
+                ArmRegOper(Rd, va=va),
+                ArmRegShiftRegOperMov(Rm, shtype, Rs),
```

Comment:
likewise, this should be three ArmRegOpers.  the mnemonic tells what kind of shift it is.